### PR TITLE
Feature/pluggable sitemaps

### PR DESCRIPTION
**Pull request by diefenbach**

### CODE REVIEW

File: docs/developer/settings.rst [hunk 1]
```diff
@@ -272,6 +272,65 @@ LFS_SEND_ORDER_MAIL_ON_PAYMENT
...
+LFS_SITEMAPS
+    A dictionary which defines several settings for sitemaps. There are four
+    sitemaps: ``category``, ``page``, ``product`` and ``shop``. Each of them can
+    have four different settings: ``sitemap``, ``changefreq``, ``priority`` and
+    ``protocol``. All sitemaps and settings are optional. If not given the
+    default settings are used, which are::
+
+     LFS_SITEMAPS = {
+         "product": {
+             "sitemap": "lfs.core.sitemap.ProductSitemap",
```
Comment (pigletto) [inline docs/developer/settings.rst]: Should be **lfs.core.sitemaps** not **lfs.core.sitemap** (same typo is present for subsequent sitemaps)

File: docs/developer/settings.rst [hunk 2]
```diff
@@ -272,6 +272,65 @@ LFS_SEND_ORDER_MAIL_ON_PAYMENT
...
+             "protocol": "http",
+         },
+         "category": {
+             "sitemap": "lfs.core.sitemap.CategorySitemap",
+             "changefreq": "weekly",
+             "priority": 0.5,
+             "protocol": "http",
+         },
+         "page": {
+             "sitemap": "lfs.core.sitemap.PagetSitemap",
```
Comment (pigletto) [inline docs/developer/settings.rst]: Should be **lfs.core.sitemaps.PageSitemap** not **lfs.core.sitemap.PagetSitemap**  (note Paget instead of Page)

File: lfs/core/urls.py [hunk 3]
```diff
@@ -35,7 +32,38 @@
...
+
+# Sitemap urls
+try:
+    product_sitemap = import_symbol(settings.LFS_SITEMAPS["product"]["sitemap"])
+except (AttributeError, KeyError, ImportError):
+    from . sitemaps import ProductSitemap
+    product_sitemap = ProductSitemap
+
+try:
+    category_sitemap = import_symbol(settings.LFS_CATEGORY_SITEMAP["categoy"]["sitemap"])
```
Comment (pigletto) [inline lfs/core/urls.py]: s/categoy/category

Comment (pigletto) [inline lfs/core/urls.py]: s/settings.LFS_CATEGORY_SITEMAP/settings.LFS_SITEMAPS

File: lfs/core/urls.py [hunk 4]
```diff
@@ -35,7 +32,38 @@
...
+except (AttributeError, KeyError, ImportError):
+    from . sitemaps import CategorySitemap
+    category_sitemap = CategorySitemap
+
+try:
+    page_sitemap = import_symbol(settings.LFS_PAGE_SITEMAP["page"]["sitemap"])
```
Comment (pigletto) [inline lfs/core/urls.py]: s/settings.LFS_PAGE_SITEMAP/settings.LFS_SITEMAPS

File: lfs/core/urls.py [hunk 5]
```diff
@@ -35,7 +32,38 @@
...
+except (AttributeError, KeyError, ImportError):
+    from . sitemaps import PageSitemap
+    page_sitemap = PageSitemap
+
+try:
+    shop_sitemap = import_symbol(settings.LFS_SHOP_SITEMAP["shop"]["sitemap"])
```
Comment (pigletto) [inline lfs/core/urls.py]: s/settings.LFS_SHOP_SITEMAP/settings.LFS_SITEMAPS

File: lfs/core/tests.py [hunk 6]
```diff
@@ -294,3 +295,82 @@ def test_manage_urls_anonymous(self):
...
+            },
+            "shop": {
+                "priority": 0.4,
+                "changefreq": "shop-daily",
+                "protocol": "shop-https",
+            },
+        }
+
+        with self.settings(LFS_SITEMAPS=my_sitemaps):
+            del sys.modules["lfs.core.sitemap"]
```
Comment (pigletto) [inline lfs/core/tests.py]: Why is this necessary? Anyway it should be "lfs.core.sitemap**s**"

Comment (pigletto) [inline lfs/core/tests.py]: If running this single test from PyCharm it fails for me with KeyError for ['lfs.core.sitemap'] and ['lfs.core.sitemaps'] as well, so if it is really required then there should be an check prior to del line, eg. if 'lfs.core.sitemap' in sys.modules:

File: lfs/core/tests.py [hunk 7]
```diff
@@ -294,3 +295,82 @@ def test_manage_urls_anonymous(self):
...
+            from lfs.core.sitemap import CategorySitemap
```
Comment (pigletto) [inline lfs/core/tests.py]: s/lfs.core.sitemap/lfs.core.sitemap**s**

applies to all these imports

**Comment by diefenbach** [comment]
Thx. All done.

File: lfs/core/tests.py (hunk 6 above)
Comment (diefenbach) [inline lfs/core/tests.py]: lfs.core.sitemaps needs to be removed before imported again. Otherwise the changes are not recognized and the test fails when running all tests.

File: lfs/core/urls.py [hunk 8]
```diff
@@ -35,7 +32,38 @@
...
+
+# Sitemap urls
+try:
+    product_sitemap = import_symbol(settings.LFS_SITEMAPS["product"]["sitemap"])
+except (AttributeError, KeyError, ImportError):
+    from . sitemaps import ProductSitemap
+    product_sitemap = ProductSitemap
+
+try:
+    category_sitemap = import_symbol(settings.LFS_SITEMAPS["categoy"]["sitemap"])
```
Comment (pigletto) [inline lfs/core/urls.py]: s/categoy/category
